source/progress.ts

```diff
@@ -3,7 +3,7 @@ import {EventEmitter} from 'events';
 import {IncomingMessage, ClientRequest} from 'http';
 import {Socket} from 'net';
 
-export function download(_response: IncomingMessage, emitter: EventEmitter, downloadBodySize?: number) {
+export function download(_response: IncomingMessage, emitter: EventEmitter, downloadBodySize?: number | null) {
```

Comment:
Let's use `undefined` instead of `null`. It's better in general to use `undefined` as it works better with modern features like default parameters. You also need to update https://github.com/sindresorhus/got/pull/708/files#diff-7bf54112e60a8714e49d3e1e98cf4e2bR15

Reply:
You needed just this change: https://github.com/sindresorhus/got/pull/708/files#diff-bc120f022c76c025170acc93be719a59R30